### DISCUSSION TITLE
Bug? Append then Insert

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#2967] [Q&A] [answered]
Hello I ran into an issue when inserting after appending removes previous data. I would think the behavior from v6.7.2 would be correct.

Steps:
- Click append, then click duplicate, first row is replaced instead of creating two items

v6.8.4
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-m0ofm?file=/src/index.js

v6.7.2
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-wihs5?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
yes, please raise a bug, looks like it's to do with `shouldUnregister` set to `false`

**Reply by fadi-george** [reply]
Will do
